Source/DafnyLanguageServer/Language/DiagnosticPublisher.cs

```diff
@@ -14,13 +14,17 @@ public class DiagnosticPublisher : IDiagnosticPublisher {
     }
 
     public void PublishDiagnostics(DafnyDocument document) {
+      if (document.LoadCanceled) {
```

Comment:
Does it ever go into the then branch? I expected a `OperationCanceledException` exception to be thrown right before the call to this method.

Reply:
Yes, the flag is used because I do not bubble the `OperationCanceledException` all the way up. The relevant locations are the following:

If a cancellation is requested, the `DocumentUpdater` falls back to the document migration mechanic. It tries to manually relocate the symbol locations. This is where the *migrate* term comes from, since I do not update the locations for the diagnostics since it's unnecessary, at least for VSCode.
https://github.com/dafny-lang/dafny/blob/2c136bbc46bff520f0f91d10349c38581b73b9fd/Source/DafnyLanguageServer/Workspace/DocumentUpdater.cs#L32-L40

The `DocumentDatabase` falls back to a placeholder document if the initial load was canceled.
https://github.com/dafny-lang/dafny/blob/2c136bbc46bff520f0f91d10349c38581b73b9fd/Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs#L65-L73

One option would be that I throw a new `OperationCanceledException` in `DocumentDatabase` to let it bubble up. But I think that this is unnecessary complexity.

Reply:
I see. I like the current design :)